packages/usa-in-page-navigation/src/index.js

```diff
@@ -42,23 +44,35 @@ const setActive = (el) => {
 };
 
 /**
- * Return an array of all visible h2 and h3 headings from the designated main content region.
- * These will be added to the component link list.
+ * Return an array of visible headings from the designated content region.
+ * These items will be added to the component link list.
  *
- * @param {HTMLElement} mainContentSelector The designated main content region
+ * @param {HTMLElement} selectedContentRegion The content region the component should pull headers from
+ * @param {String} selectedHeadingTypes The list of heading types that should be included in the link list
  *
  * @return {Array} - An array of visible headings from the designated content region
  */
-const getSectionHeadings = (mainContentSelector) => {
-  const sectionHeadings = document.querySelectorAll(
-    `${mainContentSelector} h2, ${mainContentSelector} h3`
-  );
+const getSectionHeadings = (selectedContentRegion, selectedHeadingTypes) => {
+  const headingTypes = selectedHeadingTypes.indexOf(" ")
+    ? selectedHeadingTypes.split(" ")
+    : selectedHeadingTypes;
+  const contentHeadingTypes = [];
+
+  headingTypes.forEach((headingType) => {
+    if (!IN_PAGE_NAV_VALID_HEADINGS.includes(headingType)) {
+      throw new Error(
+        `Invalid heading type: ${headingType}. Please use one or more of the following: ${IN_PAGE_NAV_VALID_HEADINGS}`
+      );
+    }
+    contentHeadingTypes.push(`${selectedContentRegion} ${headingType}`);
```

Comment:
Curious as to the need for `selectedContentRegion` here. I see `main h1,main h2,main h3,main h4,main h5,main h6`.


Do we need each heading prefixed with the content region?

Reply:
Good note. I reformatted this a bit so that we don't have to make another array in [30a3b9e](https://github.com/uswds/uswds/pull/5444/commits/30a3b9e68787790f8b133a194fd3312b52cd5682). It is similar to the approach in [this comment](https://github.com/uswds/uswds/pull/5444#discussion_r1443051902).